litt/disktable/disk_table.go

```diff
@@ -574,6 +574,71 @@ func (d *DiskTable) Get(key []byte) ([]byte, bool, error) {
 	return data, true, nil
 }
 
+func (d *DiskTable) CacheAwareGet(
+	key []byte,
+	onlyReadFromCache bool,
+) (value []byte, exists bool, hot bool, err error) {
+
+	if ok, err := d.fatalErrorHandler.IsOk(); !ok {
+		return nil, false, false, fmt.Errorf(
+			"Cannot process Get() request, DB is in panicked state due to error: %w", err)
+	}
+
+	var cacheHit bool
+	var dataSize uint64
+	if d.metrics != nil {
+		start := d.clock()
+		defer func() {
+			end := d.clock()
+			delta := end.Sub(start)
+			d.metrics.ReportReadOperation(d.name, delta, dataSize, cacheHit)
+		}()
+	}
+
+	// First, check if the key is in the unflushed data map. If so, return it from there.
+	// Performance wise, this has equivalent semantics to reading the value from
+	// a cache, so we'd might as well count it as a cache hit.
+	var rawValue any
+	if rawValue, exists = d.unflushedDataCache.Load(util.UnsafeBytesToString(key)); exists {
+		value = rawValue.([]byte)
+		cacheHit = true
+		dataSize = uint64(len(value))
+		return value, true, true, nil
+	}
+
+	// Look up the address of the data.
+	var address types.Address
+	address, exists, err = d.keymap.Get(key)
+	if err != nil {
+		return nil, false, false, fmt.Errorf("failed to get address: %w", err)
+	}
+	if !exists {
+		return nil, false, false, nil
+	}
+
+	if onlyReadFromCache {
+		// The value exists but we are not allowed to read it from disk.
+		return nil, true, false, nil
+	}
+
+	// Reserve the segment that contains the data.
+	seg, ok := d.controlLoop.getReservedSegment(address.Index())
+	if !ok {
+		return nil, false, false, nil
```

Comment:
could you comment why `!ok` means the value doesn't exist? `getReservedSegment` doc says it could also return false if reservation failed

Reply:
Is this sufficient, or would you like further explanation?

```
if !ok {
		// This can happen if there is a race between this thread and the GC thread, i.e.
		// if we start reading a value just as the garbage collector decides to delete it.
		return nil, false, false, nil
	}
```